src/dsl/lb.rs

```diff
@@ -0,0 +1,199 @@
+
+pub trait LookupBuilder<F> {
+    fn build(self, ctx: &StepTypeSetupContext<F>) -> Lookup<F>;
+}
+
+/// A helper struct for building lookup tables.
+pub struct InPlaceLookupBuilder<F> {
+    lookup: Lookup<F>,
+}
+
+impl<F> LookupBuilder<F> for InPlaceLookupBuilder<F> {
+    fn build(self, _: &StepTypeSetupContext<F>) -> Lookup<F> {
+        self.lookup
+    }
+}
+
+impl<F> Default for InPlaceLookupBuilder<F> {
+    fn default() -> Self {
+        InPlaceLookupBuilder {
+            lookup: Lookup::default(),
+        }
+    }
+}
+
+impl<F: Debug + Clone> InPlaceLookupBuilder<F> {
+    /// Adds a source column-lookup column pair to the lookup table. Because the function returns a
+    /// mutable reference to the `LookupBuilder<F>`, it can an chain multiple `add` and `enable`
+    /// function calls to build the lookup table. Requires calling `lookup` to create an empty
+    /// `LookupBuilder` instance at the very front.
+    pub fn add<C: Into<Constraint<F>>, E: Into<Expr<F>>>(
+        mut self,
+        constraint: C,
+        expression: E,
+    ) -> Self {
+        let constraint = constraint.into();
+        self.lookup
+            .add(constraint.annotation, constraint.expr, expression.into());
+        self
+    }
+
+    /// Adds a selector column specific to the lookup table. Because the function returns a mutable
+    /// reference to the `LookupBuilder<F>`, it can an chain multiple `add` and `enable` function
+    /// calls to build the lookup table. Requires calling `lookup` to create an
+    /// empty `LookupBuilder` instance at the very front.
+    pub fn enable<C: Into<Constraint<F>>>(mut self, enable: C) -> Self {
+        let enable = enable.into();
+        self.lookup.enable(enable.annotation, enable.expr);
+        self
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct LookupTableStore<F> {
+    id: UUID,
+    dest: Vec<Expr<F>>,
+}
+
+impl<F> Default for LookupTableStore<F> {
+    fn default() -> Self {
+        Self {
+            id: uuid(),
+            dest: Default::default(),
+        }
+    }
+}
+
+impl<F> LookupTableStore<F> {
+    #[allow(clippy::should_implement_trait)]
+    pub fn add<E: Into<Expr<F>>>(mut self, expr: E) -> Self {
+        self.dest.push(expr.into());
+
+        self
+    }
+
+    pub fn uuid(&self) -> UUID {
+        self.id
+    }
+}
+
+impl<F: Debug + Clone> LookupTableStore<F> {
+    pub fn build(self, src: Vec<Constraint<F>>, enable: Option<Constraint<F>>) -> Lookup<F> {
+        assert_eq!(
+            self.dest.len(),
+            src.len(),
+            "number of source and destination parameters doesn't match"
+        );
+
+        let mut lookup = Lookup::default();
+
+        if let Some(enable) = enable {
+            lookup.enable(enable.annotation, enable.expr);
+        }
+
+        src.into_iter()
+            .zip(self.dest)
+            .for_each(|(src, dest)| lookup.add(src.annotation, src.expr, dest));
+
+        lookup
+    }
```

Comment:
Will we ever build directly from a LookupTableStore instead of from a LookupBuilder using an UUID? If no, we can make this function private.

Reply:
Done